docs/developers/architecture.rst

```diff
@@ -0,0 +1,87 @@
+=========================
+ListenBrainz Architecture
+=========================
+
+Production Services
+================================
+
+Services exclusive to ListenBrainz
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+1) listenbrainz-cron-prod: runs cron jobs used to execute periodic tasks like creating dumps, invoking spark jobs to
+   import dump, requesting statistics and so on.
+
+2) listenbrainz-web-prod: runs a uwsgi server which serves ListenBrainz flask app for the website and APIs
+   (except compat APIs).
+
+3) listenbrainz-api-compat-prod: runs a uwsgi server which serves a flask app for only Last.fm compatible APIs.
+
+4) listenbrainz-api-compat-nginx-prod: ???
+
+5) listenbrainz-timescale-writer-prod: runs timescale writer which consumes listens from incoming rabbitmq queue,
+   performs a messybrainz lookup and inserts listens in the database.
+
+6) listenbrainz-websockets-prod: runs websockets server to handle realtime listen and playlist updates.
+
+7) listenbrainz-labs-api-prod: runs a uwsgi server which serves a flask app for experimental ListenBrainz APIs.
+
+8) listenbrainz-spotify-reader-prod: runs a service for importing listens from spotify API and submitting to rabbitmq.
+
+9) listenbrainz-spark-reader-prod: processes incoming results from spark cluster like inserting statistics in database etc.
+
+10) listenbrainz-redis: redis instance used for caching all stuff LB.
+
+11) exim-relay-listenbrainz.org: smtp relay used by LB to send emails.
+
+12) listenbrainz-timescale: timescale instance for LB to store listens and playlists.
+
+13) listenbrainz-typesense, listenbrainz-mbid-mapping, listenbrainz-mbid-mapping-writer-prod: ???
```

Comment:
```suggestion
13) listenbrainz-typesense, listenbrainz-mbid-mapping, listenbrainz-mbid-mapping-writer-prod: used to search for an MBID recording using the track name and artist names
```

Reply:
👍 Lets' ask @mayhem if something can be added here. Also, would be nice to specify container's role separately.